WordPress/src/test/java/org/wordpress/android/ui/activitylog/BackupDownloadProgressCheckerTest.kt

```diff
@@ -0,0 +1,144 @@
+package org.wordpress.android.ui.activitylog
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.inOrder
+import com.nhaarman.mockitokotlin2.verifyZeroInteractions
+import com.nhaarman.mockitokotlin2.whenever
+import junit.framework.Assert.assertEquals
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.TEST_SCOPE
+import org.wordpress.android.fluxc.action.ActivityLogAction.FETCH_BACKUP_DOWNLOAD_STATE
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.activity.BackupDownloadStatusModel
+import org.wordpress.android.fluxc.store.ActivityLogStore
+import org.wordpress.android.fluxc.store.ActivityLogStore.BackupDownloadStatusError
+import org.wordpress.android.fluxc.store.ActivityLogStore.BackupDownloadStatusErrorType
+import org.wordpress.android.fluxc.store.ActivityLogStore.OnBackupDownloadStatusFetched
+import java.util.Date
+
+@RunWith(MockitoJUnitRunner::class)
+class BackupDownloadProgressCheckerTest {
+    @Mock lateinit var activityLogStore: ActivityLogStore
+    @Mock lateinit var site: SiteModel
+    private val restoreId = 1L
+    private lateinit var backupDownloadProgressChecker: BackupDownloadProgressChecker
+
+    @Before
+    fun setUp() {
+        backupDownloadProgressChecker = BackupDownloadProgressChecker(activityLogStore, TEST_SCOPE)
+    }
+
+    private val finishedBackupDownload = BackupDownloadStatusModel(
+            downloadId = 1L,
+            rewindId = "rewindId",
+            backupPoint = Date(),
+            startedAt = Date(),
+            progress = null,
+            downloadCount = 0,
+            validUntil = Date(),
+            url = "url"
+    )
+
+    private val backupDownloadStatusModel = BackupDownloadStatusModel(
```

Comment:
Since the previous model is called "finishedBackDownload", we might want to consider naming this one something like "ongoing/runningBackupDownload". 

Also I don't see the purpose of `finishedBackupDownload` and `finishedBackupDownloadStatus` -> why do we need two fields when they contain the exact same values. Could we just remove `finishedBackupDownloadStatus` and use `finishedBackupDownload` instead?

Wdyt?

Reply:
Agree. I was actually considering adding the sealed classes since we don't get a progress = 100. Some of this _may_ be refactored once I get into the actual UI pieces, but for now I will get rid of it. Thanks